src/mongo/MongoTripod.class.php

```diff
@@ -657,11 +633,12 @@ protected function getDataUpdater()
     {
         if(!isset($this->dataUpdater))
         {
+            $readPreference = $this->collection->getReadPreference();
             $opts = array(
                 'defaultContext'=>$this->defaultContext,
                 OP_ASYNC=>$this->async,
                 'stat'=>$this->stat,
-                'readPreference'=>$this->collection->getReadPreference(),
+                'readPreference'=>$readPreference['type'],
```

Comment:
Should the readPreference here be set to ```$readPreference['type']``` or just ```$readPreference``` ? I've traced it through and i think it handles both, but are we ok using different values? (one with the value, one with an array that holds the value) ?